src/test/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageParserTest.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.microsoft.azure.servicebus.MessageBody;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.notificationservice.exception.InvalidMessageException;
+import uk.gov.hmcts.reform.notificationservice.model.common.ErrorCode;
+import uk.gov.hmcts.reform.notificationservice.model.request.incomming.NotificationMsg;
+
+import static com.microsoft.azure.servicebus.MessageBody.fromBinaryData;
+import static com.microsoft.azure.servicebus.MessageBody.fromSequenceData;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class NotificationMessageParserTest {
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private final NotificationMessageParser notificationMsgParser =
+        new NotificationMessageParser(objectMapper);
+
+    @Test
+    public void should_return_valid_notificationMessage_when_queue_message_is_valid() throws JSONException {
+        NotificationMsg expected = new NotificationMsg(
+            "fileName.zip",
+            "divorce",
+            "pobox",
+            "1234567890123456",
+            ErrorCode.ERR_FILE_LIMIT_EXCEEDED,
+            "size too big",
+            "orchestrator"
+        );
+
+        NotificationMsg notificationMessage =
+            notificationMsgParser.parse(
+                fromBinaryData(
+                    ImmutableList.of(
```

Comment:
Suggestion: we could have a helper method called `messageBodyFromString(String content)`, which would do these two things. Not so important.